### ISSUE TITLE
Allow output of Fluid elements in the same order/mix as they had been maintained in the UI

### ISSUE THREAD

**Issue by larsschwarz** [ExpressionEngine/ExpressionEngine#4071]
Following up this bug report here: 

https://github.com/ExpressionEngine/ExpressionEngine/issues/4067

According to this bug report it's not intended to mix regular fields and Field Groups in a Fluid at least not when it comes to ouputting these fields in a mixed and sorted order.

We use Fluids to allow users to maintain somekind of block-style maintained content. A block can be something like a headline or media content. 

For "simple" content we use regular fields, for more complex content we use Field Groups. We do provide a predefined set of blocks the user can then add to a Fluid. He can then order the blocks as he likes (using the default drag/drop functionality). 

Example of such a Fluid:

Position 1: Headline, regular (text) field
Position 2: Subline, regular (text) field
Position 3: Media, Field Group
Position 4: Copy, regular (RTE) field

The Field Group for the "Media" block contains a radio button group (to select the layout for the media that is displayed) and a Grid field (to maintain 1 or more images). 

According to the bug report there is no way to mix regular fields and Field Groups in the template. Thererefore we can only output the regular fields (Pos 1, 2 and 4) and then the Field Groups (Pos 3) or vice versa, but we cannot output the data in the order as maintained by the user in the UI. 

If there is no way to maintain the order as maintained by the user in the UI I don't understand why the UI allows mixing regular fields and Field Groups at all? From my understading the idea of the Fluid field is to add various elements and sort them as you (the user) wants them to be ordered/sorted? If this isn't the itended use then I'd expect the Fluid field to have 2 seperated areas, one for regular fields and one for Field Groups?

So I guess my feature request would simply be having option to output fields in the template in the order they had be maintained in the UI, like (**pseudo code**):

```
{fluid}
{if field_name=="field_name_or_field_group_name"}
	// output field or field_group
{if:elseif field_name=="field_name_or_field_group_name"}
	// output field or field_group
{if:elseif field_name=="field_name_or_field_group_name"}
	// output field or field_group
{if:elseif field_name=="field_name_or_field_group_name"}
	// output field or field_group
{/if}
{/fluid}
```
As a side request it would be really helpful to allow Fluid field contents to be passed to an embed. This would kinda mimic the flow that many devs know from using WP along with Gutenberg and ACF. 

More pseudo code for this aditional request:

```
{fluid}
{if field_name=="field_name_or_field_group_name"}
	{embed="block/my_block" contents="{fluid:content}"}
{if:elseif field_name=="field_name_or_field_group_name"}
	{embed="block/my_other_block" contents="{fluid:content}"}
{if:elseif field_name=="field_name_or_field_group_name"}
	{embed="block/my_other_other_block" contents="{fluid:content}"}
{if:elseif field_name=="field_name_or_field_group_name"}
	{embed="block/my_other_other_other_block" contents="{fluid:content}"}
{/if}
{/fluid}
```
Thanks

**Comment by bryannielsen** [comment]
@larsschwarz the fluid field should be maintaining the same order as the user sets in the control panel.  Maybe you could share your actual template code and we can try to have a look.  It's hard to tell if things have been left out of the pseudo-code on purpose or not but but I could see how what you have might not work.

Here's an example of what I would expect to see for your given field names.  I'm not sure what is in the Media Field Group so I wrote this example to reuse your `heading` field and add a field named `image` as well.

```
{fluid}
    {fluid:heading}
        heading - {content}
    {/fluid:heading}

    {fluid:subline}
        subline - {content}
    {/fluid:subline}

    {fluid:media}
        {fields}
            {fluid:heading}
                heading - {content}
            {/fluid:heading}

            {fluid:image}
                image - {content}
            {/fluid:image}
        {/fields}
    {/fluid:media}

    {fluid:copy}
        copy - {content}
    {/fluid:copy}
{/fluid}
```

The output of the content should not be dependent on how you order things in the template, it will be controlled by the order of the content in the control panel.

**Comment by larsschwarz** [comment]
Thanks @bryannielsen, working again on this project this week, I'll let you know about the order once I changed to respective code not to use an IF statement for the output handling.

**Comment by bryannielsen** [comment]
@larsschwarz I'm going to close this issue for now but I also wanted to mention if you are still seeing any problems I'd recommend upgrading to the latest version of EE as there are a number of fluid field improvements and fixes in 7.5 and later.

**Comment by larsschwarz** [comment]
Thanks @bryannielsen. We haven't seen any problems after we figured out the proper tag structure for outputting the field data. Now with the recently added "template tag copy/generation" functionality it's way easier and speeds up the whole process a lot!
